packages/libs/src/sdk/services/Solana/programs/BaseSolanaProgramClient.ts

```diff
@@ -179,4 +186,24 @@ export class BaseSolanaProgramClient {
       })
     )
   }
+
+  /**
+   * Normalizes the instructions as TransactionInstruction whether from
+   * versioned transactions or legacy transactions.
+   */
+  protected async getInstructions(
+    transaction: VersionedTransaction | Transaction
+  ) {
+    if (!('instructions' in transaction)) {
```

Comment:
nit: is this the best way to do this? typeguard?

Reply:
probably but i'd leave the truthy case first

Reply:
yeah, we can't do `instanceof` or we break things in React Native I'm pretty sure. Can flip the if statement though